src/value.c

```diff
@@ -76,11 +82,28 @@ SIValue SI_TransferStringVal(char *s) {
 	};
 }
 
+// deep clone an array of SIValues
+SIValue SI_CloneArrayVal(SIValue *array) {
+	size_t arrayLen = array_len(array);
+	SIValue *clone = array_new(SIValue, arrayLen);
+	for(int i = 0; i < arrayLen; i++) {
+		SIValue clonedValue = SI_Clone(array[i]);
+		array_append(clone, clonedValue);
```

Comment:
Since you've sized `clone` properly, this is a bit overcautious, but I think it's best to always use the construction
`clone = array_append(clone, clonedValue);`
Which is realloc-safe.

Reply:
fixed